### PR TITLE
Fix Databus for NewtonsoftJsonSerializer

### PR DESCRIPTION
**Pull request by jpalac**
Fixes: https://github.com/Particular/NServiceBus/issues/6675

However, discovered an issue on .Net8 (only) with NServiceBus.Newtonsoft.Json where the Value property of the DataBusProperty class gets serialized.

### CODE REVIEW

File: src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj [hunk 1]
```diff
@@ -12,6 +12,7 @@
   <ItemGroup>
     <PackageReference Include="GitHubActionsTestLogger" Version="2.3.3" PrivateAssets="All" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0-release-23468-02" PrivateAssets="All" />
+    <PackageReference Include="NServiceBus.Newtonsoft.Json" Version="3.0.0" />
```
Comment (bording) [inline src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj:15]: Core should not have any references to downstream packages, so you can't add this package reference.

Comment (jpalac) [reply src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj:15]: @bording  Even to test projects?

Comment (bording) [reply src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj:15]: Yes, even test projects. The AcceptanceTests project produces a source package that is used downsteram.

Even for test projects in Core that don't produce packages, adding a downstream package adds a transitive reference to the Core package, which means you now have a Core project referencing itself.

Comment (jpalac) [reply src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj]: Removed the reference and hence the test.

**Comment by jpalac** [comment]
Closing as superseded by https://github.com/Particular/NServiceBus/pull/6894

**Pull request closed without merging**